ruby/lib/ci/queue/redis/base.rb

```diff
@@ -44,6 +44,55 @@ def initialize(redis_url, config)
           end
         end
 
+        class Monitor
+          def initialize(redis_url, zset_key, processed_key)
+            @redis_url = redis_url
+            @zset_key = zset_key
+            @processed_key = processed_key
+          end
+
+          def boot!
+            child_read, @pipe = IO.pipe
+            ready_pipe, child_write = IO.pipe
+            @pipe.binmode
+            @pid = Process.spawn(
+              RbConfig.ruby,
+              ::File.join(__dir__, "monitor.rb"),
+              @redis_url,
+              @zset_key,
+              @processed_key,
+              in: child_read,
+              out: child_write,
+            )
+            child_write.close
+            child_read.close
+
+            # Check the process is alive.
+            ready_pipe.wait_readable(2)
+            Process.kill(0, @pid)
+            @pipe
+          end
+
+          def shutdown!
+            @pipe.close
```

Comment:
We  should wait for the child here, with `Process.wait(@pid)`.

It's not strictly necessary, because `pid` would do it for us, but it's much cleaner.